samples/node/routes/discovery.js

```diff
@@ -5,18 +5,17 @@
 
 "use strict";
 
-const hal = require('halberd');
+const fs = require('fs');
 
 exports.root = function (req, res) {
-    const resource = new hal.Resource();
     const base = `${protocol(req)}://${host(req)}`;
-    resource.link('metadata', `${base}/discovery/metadata.json`);
-    resource.link('objects', `${base}/cards/requests`);
-    resource.link('image', `${base}/images/connector.png`);
-    resource.link('test_auth', `${base}/test-auth`);
-    res.setHeader('Content-Type', 'application/hal+json');
-    res.send(resource.toJSON());
-}
+    fs.readFile('./discovery/metadata.json', 'utf8', function (err, contents) {
```

Comment:
Why not just move the metadata.json into this function?

```
const base = `${protocol(req)}://${host(req)}`;
const body = {
  image: {href: `${base}/images/connector.png`},
  test_auth: {href: `${base}/test-auth`},
  object_types: [
    {
      name: "card",
      doc: {href: "https://github.com/vmwaresamples/card-connectors-guide/wiki/Card-Responses"},
      fields: {
        zip: {capture_group: 1, regex: "([0-9]{5})(?:[- ][0-9]{4})?"}
      },
      endpoint: {href: `${base}/cards/requests`}
    }
  ]
};
return res.status(200).json(body);
```

Reply:
Ya, this seems better than reading from file every time.